cabal-install/Distribution/Client/ProjectConfig/Legacy.hs

```diff
@@ -1098,6 +1119,22 @@ legacyPackageConfigFieldDescrs =
       , "fail-when-no-test-suites", "test-wrapper" ]
   . commandOptionsToFields
   ) (testOptions' ParseArgs)
+ ++
+  ( liftFields
+      legacyBenchmarkFlags
+      (\flags conf -> conf { legacyBenchmarkFlags = flags })
+  . mapFieldNames
+      prefixBenchmark
```

Comment:
This is redundant since all your options already have the `benchmark-` prefix. The reason the `test` counterpart has this is because there we let unprefixed options through with filterFields.